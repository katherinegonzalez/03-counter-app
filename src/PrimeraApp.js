// Functional Component FC
const PrimeraApp = () => {
    return ( // Siempre deben estar contenidos en un solo elemento para hacer el return, puede ser un div o un Fragment
        <div>
            <h1>Hola Mundo</h1>
            <p>Mi primera aplicaci√≥n</p>
        </div>  
    );
        
};

export default PrimeraApp;


