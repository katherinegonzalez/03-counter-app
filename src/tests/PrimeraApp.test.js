// import { render } from "@testing-library/react";
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import PrimeraApp from "../PrimeraApp";


describe('Pruebas en <PrimeraApp />', ()=>{

    /*test('Debe mostrar el mensaje "Hola, Soy Goku"', () => {
        const saludo = 'Hola, Soy Goku';

        const { getByText } = render( <PrimeraApp saludo={ saludo } otra={123} /> );
        
        // expect( getByText(saludo)).toBeInTheDocument(); para que esta funcione debemos hacer el import '@testing-library/jest-dom/extend-expect' en el setupTest.js;
    });*/

    test('Debe mostrar <PrimeraApp /> correctamente', () => {
        
        const saludo = 'Hola, Soy Goku';
        const wrapper = shallow( <PrimeraApp saludo={ saludo } otra={123}/> );
        
        expect (wrapper).toMatchSnapshot();
    })

    test('Debe enviar el subtÃ­tulo enviado por props', () => {
        
        const saludo = 'Hola, Soy Goku';
        const subtitulo = 'Soy un subtitulo';

        const wrapper = shallow( 
            <PrimeraApp 
                saludo={ saludo } 
                otra={123}
                subtitulo ={subtitulo}
                />
            );
        
            const textoParrafo = wrapper.find('#subtitulo').text();
            console.log(textoParrafo);

            expect(textoParrafo).toBe(subtitulo);

    })
    

});