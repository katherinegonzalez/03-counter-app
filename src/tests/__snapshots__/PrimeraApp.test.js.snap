// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> Debe mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola, Soy Goku
  </h1>
  <p
    id="subtitulo"
  >
    Soy un subtitulo
  </p>
  <p>
    Mi primera aplicaci√≥n
  </p>
</Fragment>
`;
